app/uk/gov/hmrc/cdsimportsddsfrontend/controllers/forms/DocumentationTypeFormMapping.scala

```diff
@@ -38,14 +38,21 @@ object AdditionalPaymentTypeFormMapping {
   )(AdditionalPaymentType.apply)(AdditionalPaymentType.unapply))
 }
 
+object PreviousDocumentFormMapping {
+  val previousDocument: (String, Mapping[Seq[PreviousDocument]]) = "previousDocument" -> seq(mapping(
+    "categoryCode" -> optional(text),
+    "id" -> optional(text),
+    "typeCode" -> optional(text),
+    "lineNumeric" -> optional(text)
+  )(PreviousDocument.apply)(PreviousDocument.unapply))
+}
+
 object DocumentationTypeFormMapping {
   import AdditionalPaymentTypeFormMapping.paymentType
   import AdditionalDocumentFormMapping.additionalDocumentType
+  import PreviousDocumentFormMapping.previousDocument
   val documentationType: (String, Mapping[DocumentationType]) = "documentationType" -> mapping(
-    "previousDocCategory" -> optional(text),
-    "previousDocType" -> optional(text),
-    "previousDocReference" -> optional(text),
-    "previousDocGoodsItemId" -> optional(text),
+    previousDocument,
```

Comment:
It would be clearer to leave the key (`"previousDocument" -> ...`)at this point in the code

Reply:
I feel this lets you know you're dealing with a nested form.

Reply:
i'm ok with this, since it's a nested form.  this sort of thing will largely go away in the end when we have multiple pages instead of one huge form.